controller/controller.go

```diff
@@ -1404,10 +1404,14 @@ func (ctrl *ProvisionController) provisionClaimOperation(ctx context.Context, cl
 	if nodeName, ok := getString(claim.Annotations, annSelectedNode, annAlphaSelectedNode); ok {
 		if ctrl.nodeLister != nil {
 			selectedNode, err = ctrl.nodeLister.Get(nodeName)
+			// if node does not exist, remove volume.kubernetes.io/selected-node annotation
+			if apierrs.IsNotFound(err) {
+				delete(claim.Annotations, annSelectedNode)
+			}
 		} else {
 			selectedNode, err = ctrl.client.CoreV1().Nodes().Get(ctx, nodeName, metav1.GetOptions{}) // TODO (verult) cache Nodes
 		}
-		if err != nil {
+		if err != nil && !apierrs.IsNotFound(err) {
```

Comment:
+1, when the node does not exist, the controller should go to `ctrl.rescheduleProvisioning` path and send events and whatnot. It may need some bigger refactoring to get to that code path easily.

Reply:
So if there is NotFound error, we set the result == ProvisioningReschedule? and then call ctrl.rescheduleProvisioning?